platform_plugin_turnitin/turnitin.py

```diff
@@ -0,0 +1,426 @@
+        """
+        Submits acceptance of the EULA for the current user.
+
+        Args:
+            data (dict): Input data for the request.
+            suffix (str, optional): Additional suffix for the request. Defaults to ''.
+
+        Returns:
+            dict: The response after accepting the EULA.
+        """
+        user_id = self.get_user_data()["user_id"]
+        date_now = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+        payload = {
+            "user_id": str(user_id),
+            "accepted_timestamp": date_now,
+            "language": "en-US",
+        }
+        response = post_accept_eula_version(payload)
+        return response.json()
+
+    def create_turnitin_submission_object(self):
+        """
+        Constructs a Turnitin submission object based on the user's data.
+
+        Returns:
+            Response: The response from the Turnitin submission API.
+        """
+        user_data = self.get_user_data()
+        user_name = user_data["user_name"]
+        date_now = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+        payload = {
+            "owner": user_data["user_id"],
+            "title": self.location.block_id,
+            "submitter": user_data["user_id"],
+            "owner_default_permission_set": "LEARNER",
+            "submitter_default_permission_set": "INSTRUCTOR",
+            "extract_text_only": False,
+            "metadata": {
+                "owners": [
+                    {
+                        "id": user_data["user_id"],
+                        "given_name": user_name[0] if user_name else "no_name",
+                        "family_name": " ".join(user_name[1:])
+                        if len(user_name) > 1
+                        else "no_last_name",
+                        "email": user_data["user_email"],
+                    }
+                ],
+                "submitter": {
+                    "id": user_data["user_id"],
+                    "given_name": user_name[0] if user_name else "no_name",
+                    "family_name": " ".join(user_name[1:])
+                    if len(user_name) > 1
+                    else "no_last_name",
+                    "email": user_data["user_email"],
+                },
+                "original_submitted_time": date_now,
+            },
+        }
+        return post_create_submission(payload)
+
+    @XBlock.handler
+    def upload_turnitin_submission_file(self, data, suffix=""):
+        """
+        Handles the upload of the user's file to Turnitin.
+
+        Args:
+            data (WebRequest): Web request containing the file to be uploaded.
+            suffix (str, optional): Additional suffix for the request. Defaults to ''.
+
+        Returns:
+            Response: The response after uploading the file to Turnitin.
+        """
+        turnitin_submission = self.create_turnitin_submission_object()
+        if turnitin_submission.status_code == 201:
```

Comment:
Can we use the [HTTP Status Code](https://docs.python.org/3/library/http.html) package?